xpallets/gateway/bitcoin/v2/src/vault.rs

```diff
@@ -87,7 +88,16 @@ pub mod pallet {
     pub struct Pallet<T>(PhantomData<T>);
 
     #[pallet::hooks]
-    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_finalize(_: BlockNumberFor<T>) {
+            if assets::Pallet::<T>::is_bridge_running() {
+                <Vaults<T>>::translate(|_, vault| {
```

Comment:
Why `translate`?

Reply:
Cause may mutate vault status

Reply:
`translate` is mostly used when doing the migration, if you want to mutate a storage item, just use `mutate`.